roles/httpd/tasks/main.yml

```diff
@@ -1,21 +1,13 @@
 ---
-- name: Install Apache httpd
-  ansible.builtin.package:
-    name:
-      - httpd
-      - mod_ssl
+- name: Pull Apache container image
+  containers.podman.podman_image:
+    name: "{{ httpd_container_image }}"
     state: present
 
-- name: Set httpd_can_network_connect so Apache can connect to Puma and Gunicorn
-  ansible.posix.seboolean:
-    name: httpd_can_network_connect
-    state: true
-    persistent: true
-
-- name: Disable welcome page
-  ansible.builtin.file:
-    path: /etc/httpd/conf.d/welcome.conf
-    state: absent
+- name: Disable welcome page in container
+  containers.podman.podman_container_exec:
+    container: httpd
+    command: rm -f /etc/httpd/conf.d/welcome.conf
```

Comment:
Looking at the container definition it actually uses `/usr/local/apache2/conf.d` so this shouldn't make a difference. I also wonder if such an exec persist or whether it's needed at all. We mount a volume over `conf.d` so that volume should determine `welcome.conf`.

Reply:
After testing, I can confirm that container uses `/etc/httpd/conf.d` for its configuration. The mounted volume does not exclude `welcome.conf`, so removing it explicitly is necessary.
However, I understand that the removal via podman exec will not persist across restarts. Would modifying the host-mounted volume to exclude the file entirely be a better approach here?

Reply:
I think it highlights a potential problem with mounting those directories: we don't know what's in there. In this case you probably installed httpd which created the file.

This makes me wonder if we should create our own directory in `/etc`, like `/etc/foreman-quadlet/httpd` for Apache

Reply:
> This makes me wonder if we should create our own directory in /etc, like /etc/foreman-quadlet/httpd for Apache

I tried doing it, you can see my new changes but the problem I'm facing is how can we make sure that container state persist. I see Podman containers do not automatically persist their state.

Reply:
I meant to create the directory on the host, not in the container itself.